client/components/command-pallette/README.md

```diff
@@ -0,0 +1,69 @@
+
+In order to add a new command to the palette, you can include the command object in `useCommandsArrayWpcom` or create a new hook and include it to `useCommandPallette` hooks.
+
+## Command properties
+
+Every command has a set of properties that can be used to customize its behavior. The following properties are available:
+
+- `name`: Type `string`, used as identifier for the command. It's expected to be unique and `camelCase`. It's not visible for the user.
+- `label`: Type `string`, used as the label for the command. Visible in the command palette.
+- `subLabel`?: Type `string`, used as a second line for the command. Visible in the command palette.
+- `searchLabel`?: Type `string`, used to match the user search. It's not visible for the user. We recommend to use `_x()` to add context to the translators.
+- `callback`: Type `function`, used to execute the command when the user clicks on it or presses enter on that command.
+- `context`?: Type `string[]`, a list of URL paths for which the command will have priority when displayed. Example: `['/sites', '/manage/domains']`.
+- `icon`?: Type `JSX.Element`, used to display an icon for the command. It's visible in the command palette.
+- `image`?: Type `JSX.Element`, used to display an image for the command, instead of an icon. It's visible in the command palette.
+- `siteFunctions`?: Type `object`, used for nested commands that need to execute a function when a site is selected in the command palette as a second step. The behaviour for nested commands is the same independently of the page we are.
+
+## Best practices for defining a command
+
+Please consider carefully before adding a new command to the palette. We aim to avoid incorporating commands that are infrequently used and unlikely to be removed in the future.
+
+- Command labels should be short and descriptive and use sentence case. For example, "Add new site" instead of "Add New Site".
+- Commands should start with a verb. For example, "Add new site" instead of "New site".
+- Picking the correct verb is important. `Open` is the more generic verb, if the page has a list of elements, we can use `View`, if the user is expected to take an action we can use `Manage` or `Change`. We prefer specific verbs over generic ones.
+- We prefer to use [Gutenberg icons](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) for root commands.
+- It's important to keep the same terminology and icons used elsewhere in the interace or in core.
+- We can introduce aliases, and keywords using `searchLabel`. For example. "Import a site" would be an alias of "Migrate a site" because the navigation it's under Tools → Import.
+- Most commands navigate to other pages. We can also navigate to a specific section/card inside a page. For example, "Manage cache settings" in addition to "Manage hosting configuration".
+- Commands that open modals are also allowed. Just be aware that the command palette is visible in most Calypso pages.
```

Comment:
> Just be aware that the command palette is visible in most Calypso pages.

What are the implications of this? That the modal will need to be loaded on those pages too?

Reply:
Yes, that's what would happen by default.
I added a comment to consider loading the modal async https://github.com/Automattic/wp-calypso/pull/84591/commits/2ac72b883479fdeca1ca9ce6a0c72846868fa2c9